tensorboard/webapp/runs/views/runs_table/runs_table_test.ts

```diff
@@ -711,6 +701,53 @@ describe('runs_table', () => {
         expect(saveButton!.textContent).toContain('Save');
       });
 
+      it('dispatches `runGroupByChanged` on clicking `Regex` when there is a regex string', () => {
+        store.overrideSelector(getEnabledColorGroup, true);
+        store.overrideSelector(getEnabledColorGroupByRegex, true);
+        store.overrideSelector(getRunGroupBy, {key: GroupByKey.EXPERIMENT});
+        store.overrideSelector(getColorGroupRegexString, 'run');
+        const fixture = createComponent(
+          ['book'],
+          [RunsTableColumn.RUN_NAME, RunsTableColumn.RUN_COLOR]
+        );
+        fixture.detectChanges();
+
+        const menuButton = fixture.debugElement
+          .query(By.directive(RunsGroupMenuButtonContainer))
+          .query(By.css('button'));
+        menuButton.nativeElement.click();
+
+        getColorGroupByHTMLElement(GroupByKey.REGEX)!.click();
+        expect(dispatchSpy).toHaveBeenCalledWith(
+          runGroupByChanged({
+            experimentIds: ['book'],
+            groupBy: {key: GroupByKey.REGEX, regexString: 'run'},
+          })
+        );
+      });
+
+      it('opens edit dialog on clicking `Regex` when the regex string is not set', () => {
+        store.overrideSelector(getEnabledColorGroup, true);
+        store.overrideSelector(getEnabledColorGroupByRegex, true);
+        store.overrideSelector(getRunGroupBy, {key: GroupByKey.EXPERIMENT});
```

Comment:
Please add `store.overrideSelector(getColorGroupRegexString, '');` explicitly instead of relying on `beforeEach` or previous spec.